### DISCUSSION TITLE
Resource Limitations and Reservations

### DISCUSSION THREAD

**Discussion by QarthO** [coollabsio/coolify#4661] [Feature Requests]
I would love it if in the UI there was a built in manager to specific resource limits and reservations for services. https://docs.docker.com/reference/compose-file/deploy/#resources

Basically instead of having to manually go into the docker compose file and manually add these limits and reservations they could be sliders on the ui.

Most importantly there should be a reservation for Coolify itself. I came across an issue where I installed a service using a docker image that created a Java Virtual Machine, which by default allocated 1gb of ram. My machine while it did have more than 1gb of ram, it didnt have that much available due to the other services (including coolify) running on it. This caused the entire machine to basically brick itself. Every other service was timing out. I was also unable to SSH to the machine. Rebooting didn't fix anything either because coolify just tried to spin up the instance again. I was able to fix it by rescaling the machine to give it more memory, then adjust the service so it didnt use so much memory.

Basically, the entire goal of this feature is to have resource limits and reservations assignments easily available on every service via the ui. Along with having a default reservation for coolify itself. Simply adding a service shouldnt be able to so easily brick the entire machine. If coolify reserved some resources to the OS and to itself this would prevented services from eating into it.

**Comment by alepouna** [comment]
Seconding this

**Comment by M1ck0** [comment]
This feature would be cool, but also you made a mistake. Here is 'Note' from the installation docs

> It is strongly recommended to use a fresh, newly created server for installing Coolify. Installing on an existing server with other applications may lead to conflicts or unexpected behavior.

**Reply by QarthO** [reply]
What? There were no other applications on the server. It was a fresh server. Everything I installed was directly thru Coolify. Not sure what you're saying here

**Reply by M1ck0** [reply]
Coolify should always have server for itself. No other resources on it

**Reply by QarthO** [reply]
You're just not understanding what's going on here. Please re-read everything I've stated. Every single service that is on the machine is thru coolify. Coolify is and was the first and only thing installed on machine. I hope this makes sense to you. Please take a chance to read. If that's still not making sense I'd recommend you research what Coolify exactly is

**Reply by HomingHamster** [reply]
Resource usage statistics for the docker containers for the apps managed by coolify (such as keeping containers up to date when new containers are released and seeing how much bandwidth or cpu used hopefully), thats what im hoping for

**Reply by Bink-lab** [reply]
> Coolify should always have server for itself. No other resources on it

that is the biggest bs ive ever heard

**Comment by AkritW** [comment]
Would love to see this features!

**Comment by HomingHamster** [comment]
me too

**Comment by lukas-walker** [comment]
I would love this!

**Comment by sovetski** [comment]
Super idea!

**Comment by flyos777** [comment]
would love it too !

**Comment by jofflin** [comment]
+1
